gitops/core.py

```diff
@@ -129,22 +129,31 @@ def untag(ctx, filter, tag, exclude=''):
 
 
 @task
-def getenv(ctx, filter, exclude='', **kwargs):
-    """ Set one or more env vars on selected app(s).
-        However, please make more broad-reaching environment changes at the chart level though.
+def getenv(ctx, filter, exclude='', keys='', **kwargs):
+    """ Get one or more env vars on selected app(s).
     """
-    # TODO: FINISH THIS
-    # - offer flag to list ALL vars (from global env config too?)
-    # - tabulate output nicely, or just list one after another.
-    raise NotImplementedError
-    try:
-        apps = get_apps(filter=filter, exclude=exclude, message=f"{colourise('The env var(s)', Fore.LIGHTBLUE_EX)}\n{colourise(kwargs, Fore.LIGHTYELLOW_EX)}\n{colourise('will be added to the following apps:', Fore.LIGHTBLUE_EX)}")
-    except AppOperationAborted:
-        print(success_negative('Aborted.'))
-        return
+    get_value('environment', filter, exclude, keys, **kwargs)
+
+
+@task
+def getsecrets(ctx, filter, exclude='', keys='', **kwargs):
+    """ Get one or more secrets on selected app(s).
+    """
+    get_value('secrets', filter, exclude, keys, **kwargs)
+
+
+def get_value(value_type, filter, exclude, filter_values, **kwargs):
```

Comment:
This method name is a little weak. I'd lean towards something like `_getenv`. Feels wrong as it doesn't love both of its children equally, but secrets are env vars too..

Reply:
Agreed.